### ISSUE TITLE
Knives have absolutely insane DPS

### ISSUE THREAD

**Issue by PancakeIdentity** [MysticMods/MysticalWorld#114]
### General Information

**Describe the bug:**
The knives have extremely high DPS, making them the best choice of weapon over an axe or sword in almost every case. For example, cactus knives have a DPS of 13.5, while cactus swords have a DPS of 5.6 and axes have a DPS of 4.95. For reference, diamond swords have a DPS of 11.2. Diamond knives have a DPS of 19.6 as well. Those are just a few examples, most other knives also have this issue.

This creates a huge imbalance issue as knives have no downside to them such as lower durability or lower reach, both of which could help balance knives (though I still think the DPS difference would be too large).

**Expected behavior:**
The knives should have DPS that makes sense with the DPS of the other ~~tools~~ weapons.

***

### Environment Versions
mysticalworld-1.14.4-1.6.12

**Mystic Mods Versions**
- MysticalLib: mysticallib-1.14.4-1.10.0

**Other Versions:**
- Forge: forge-28.1.109
- Minecraft: 1.14.4

**Comment by noobanidus** [comment]
How are you measuring DPS? This isn't really a quantifiable value when it comes to programming: attack speed and attack damage are values that have already been adjusted with knives before in the past. Perhaps you could clarify your numbers? How are you measuring DPS? Are you suggesting reduction in damage, or reduction in attack speed, or both?

Unfortunately, lower reach is not something that can easily be implemented (likewise the spears don't function the way that I wanted them to for that reason). 

The feedback is greatly appreciated, I'm just struggling to understand it because DPS is a term I associate with MMORPGS like WoW, whereas the entirety of the time that I've played Minecraft people have only ever referred specifically to the flat damage value of tools/weapons.

(Also, just to clarify, knives in 1.14 are meant to be a fast weapon, not actually a tool.)

**Comment by PancakeIdentity** [comment]
How many times you can swing your weapon per second is defined by `Attack Speed`, so DPS would be `Damage`\*`Attack Speed`. The minecraft wiki uses this calculation and it holds true in testing. For example, the cactus knife deals 4.5 damage and has an attack speed of 3, meaning it does 4.5*3, or 13.5, damage per second.

This means if you swing as soon as the cooldown is done, you can do, for example, 13.5 damage to a mob per second, making it by far the most efficient weapon-type in the game.

(DPS is also why axes are still generally worse than swords, they attack slower so they have a lower damage output over time, despite having higher damage per swing)

I didn't really have a suggested fix, my initial thought was reduced range (on top of some rebalancing), but I think lower durability could help. As for which value to reduce, I'm not sure. That seems more like a problem for mod designers to solve. I'd be inclined to decrease damage, as the knife's whole gimmick is attacking quickly. Maybe slightly reducing the attack speed could help as well.

(And yeah, I meant to say weapon, not tool, at the end there)

**Comment by noobanidus** [comment]
Ah, thanks. I don't really hang out with the Minecraft wiki and I haven't run across the term before. That's a good calculation though, it means I can apply them to stuff.

**Comment by noobanidus** [comment]
Part of the confusion is that attack speed is determined using a value that is completely different from what is actually displayed in-game, which makes things a little bit of a nightmare. I've had to dig around a little to understand how the system works -- it's basically "the more you subtract the slower it is". 

From looking at how tools handle damage, I think knives were just being given too much of a modifier -- 2.5 plus the base of the tool material type. Cactus was certainly not meant to do that much damage, I can assure you. I've narrowed down the ranges and will test them out to see if that's an improvement.
